packages/components/src/components/tag/tag-utils.ts

```diff
@@ -16,6 +17,29 @@ export const hasInvertedThemeColor = (tagColor: TagColor, theme: Theme): boolean
   const isDark = isThemeDark(theme);
   return (
     (!isDark && tagColor === 'neutral-contrast-high') ||
-    (isDark && tagColor !== 'background-surface' && tagColor !== 'background-default')
+    (isDark &&
+      tagColor !== 'background-surface' &&
+      tagColor !== 'background-default' &&
+      tagColor !== 'notification-information' &&
+      tagColor !== 'notification-warning' &&
+      tagColor !== 'notification-success' &&
+      tagColor !== 'notification-error')
   );
 };
+
+export const getThemedBackgroundHoverColor = (tagColor: TagColor, themedColors: ThemedColors, theme: Theme): string => {
+  const isDark = isThemeDark(theme);
+  const colorMap: { [key in TagColor]: string } = {
+    'background-default': isDark ? themedColors.backgroundColorLighten : themedColors.backgroundColorDarken,
+    'background-surface': isDark
+      ? themedColors.backgroundSurfaceColorLighten
+      : themedColors.backgroundSurfaceColorDarken,
+    'neutral-contrast-high': isDark ? themedColors.contrastHighColorLighten : themedColors.contrastHighColor,
```

Comment:
I think `contrastHighColor` should be `contrastHighColorDarken` or why break the pattern?

Reply:
@denyo for "contrast high" color tag we use "primary" color as background, it's almost black for light theme (#010205). The reason for that is that Janick and Johannes told that the "contrastHighColor" from our design tokens is not black enough for "contrast hight tag".
So since  "contrast high" tag is already very black, we can't make it darker when hovering over it (or we can, but the user won't notice it). So we make it lighter when hovering over it  (contrastHighColor is lighter as primaryColor) - it's an exception from the rule.